compiler/rustc_trait_selection/src/solve/trait_goals.rs

```diff
@@ -599,7 +599,7 @@ impl<'tcx> assembly::GoalKind<'tcx> for TraitPredicate<'tcx> {
         // which will ICE for region vars.
         let args = ecx.tcx().erase_regions(goal.predicate.trait_ref.args);
 
-        let Some(assume) =
+        let Some(Ok(assume)) =
```

Comment:
given that there should still be ambiguity here, we should probably fully match on the returned value